.github/workflows/linting.yml

```
@@ -0,0 +1,38 @@
+name: Linting
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    paths:
+      - '.github/workflows/*.yml'
+  workflow_dispatch:
+
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Checkout code
+        with:
+          persist-credentials: false
+        uses: actions/checkout@v4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install zizmor==0.9.2
+
+      # Apply GitHub Actions linter, zizmor
+      - name: zizmor
+        if: always()
```

```suggestion
        if: ${{ !cancelled() }}
```

It's recommended to avoid `if: always()` - see https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#always . What I linked recommends we use `!cancelled()`